### PR TITLE
refactor: refinement of budgets list based on previous pr #P23-322

### PR DESCRIPTION
**Pull request by AzulloCyll**
Previous PR of this task is here https://github.com/intive/patronage2023-js/pull/85

What was done:
1. Removed delay from useInfiniteQuery queryFn
2. Removed axios and used fetch in `getBudgetsList`

### CODE REVIEW

File: apps/web/app/(navigation)/SideNavigationBar.tsx
Comment (bdeluga) [inline apps/web/app/(navigation)/SideNavigationBar.tsx:1]: - I feel like there are too many return statements that can be omitted. 
- We could provide some types for next maintainers of our code, and we can also omit using `any`

For example:
```tsx


interface Item {
  id: {
    value: string;
  };
  icon: IconType;
  name: string;
}

interface InifityScrollResponse {
  items: Item[];
}

  const successData =
    data?.pages?.flatMap(({ items }: InifityScrollResponse) =>
      items.map(({ icon, name, id }) => ({
        ComponentToRender: (
          <>
            <IconStyled
              icon={iconNames.includes(icon) ? icon : "help"}
              iconSize={24}
            />
            <SpanStyled>{name}</SpanStyled>
          </>
        ),
        href: `/budgets/${id.value}`,
        id: id.value,
        ref: lastBudgetRef,
      }))
    ) ?? [];

```

Then we could pass contents to `NavList` as `contents={successData}`

**Review by bdeluga — CHANGES_REQUESTED** [review]
Small thing to debate on. 👀

File: apps/web/app/(navigation)/SideNavigationBar.tsx
Comment (AzulloCyll) [reply apps/web/app/(navigation)/SideNavigationBar.tsx:1]: done

File: apps/web/services/mutations.ts [hunk 1]
```diff
@@ -19,36 +18,37 @@ export type GetBudgetsListType = {
...
+const URL = env.NEXT_PUBLIC_API_URL + "budgets/list";
 
 export const getBudgetsList = async ({
   pageSize,
   pageParam,
   searchValue,
   sortAscending,
-  axiosInstance,
+  token,
 }: GetBudgetsListType) => {
```
Comment (Noshi96) [inline apps/web/services/mutations.ts:32]:
```suggestion
}: GetBudgetsListType): Promise<ItemType> => {
```

File: apps/web/lib/hooks/useGetBudgets.tsx [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
   sortAscending: boolean,
   pageSize: number
 ) => {
   const { data: sessionData } = useSession();
-  const axiosInstance = reqInstance(sessionData?.user.accessToken);
+  const token = sessionData?.user.accessToken;
 
   return useInfiniteQuery({
     queryKey: ["budgets", { searchValue, sortAscending }],
     queryFn: async ({ pageParam = 1 }) => {
```
Comment (Noshi96) [inline apps/web/lib/hooks/useGetBudgets.tsx:16]:
```suggestion
    queryFn: async ({ pageParam = 1 }): Promise<ItemType> => {
```

File: apps/web/app/(navigation)/SideNavigationBar.tsx [hunk 3]
```diff
@@ -70,24 +83,25 @@ export default function SideNav() {
...
-        ),
-        href: `/budgets/${item.id.value}`,
-        id: item.id.value,
-        ref: lastBudgetRef,
-      };
-    });
-  });
+  const successData =
+    data?.pages?.flatMap(
+      ({ items }: InifityScrollResponse) =>
```
Comment (Noshi96) [inline apps/web/app/(navigation)/SideNavigationBar.tsx:88]:
```suggestion
      ({ items }) =>
```

**Review by Noshi96 — COMMENTED** [review]
What do you think about declaring these types for the returned data in the fetch and react query functions 🤔?

File: apps/web/app/(navigation)/SideNavigationBar.tsx
Comment (AzulloCyll) [reply apps/web/app/(navigation)/SideNavigationBar.tsx:1]: You are absolutly right :)

**Comment by AzulloCyll** [comment]
> What do you think about declaring these types for the returned data in the fetch and react query functions 🤔?

I think its goot to have it done. I'll change theese later today.

**Review by bdeluga — COMMENTED** [review]
LGTM 🚀

**Review by bdeluga — APPROVED** [review]

File: apps/web/app/(navigation)/SideNavigationBar.tsx (hunk 3 above)
Comment (AzulloCyll) [reply apps/web/app/(navigation)/SideNavigationBar.tsx]: done

**Review by monika-klokosinska-intive — APPROVED** [review]

**Pull request merged**